### DISCUSSION TITLE
Python Segmentation Fault, ../sysdeps/unix/sysv/linux/raise.c: No such file or directory

### DISCUSSION THREAD

**Discussion by ilDeffo** [pyinstaller/pyinstaller#5927] [Help]
I created a single file executable of my GUI app with PyInstaller. Everything looks fine, the executable is created and starts. First, the login page is correctly shown, but after inserting username and password the program crashes. I tried to debug with gdb, and this is what I get:

    Program received signal SIGSEGV, Segmentation fault.
    raise (sig=<optimized out>) at ../sysdeps/unix/sysv/linux/raise.c:50
    50	../sysdeps/unix/sysv/linux/raise.c: No such file or directory.
    (gdb) bt
    #0  raise (sig=<optimized out>) at ../sysdeps/unix/sysv/linux/raise.c:50
    #1  0x000055555555a213 in ?? ()
    #2  0x000055555555815d in ?? ()
    #3  0x00007ffff7da10b3 in __libc_start_main (main=0x5555555564d0, argc=1, 
        argv=0x7fffffffdec8, init=<optimized out>, fini=<optimized out>, 
        rtld_fini=<optimized out>, stack_end=0x7fffffffdeb8)
        at ../csu/libc-start.c:308
    #4  0x000055555555650a in ?? ()

Can someone explain what does it mean and how to solve?

**Comment by rokm** [comment]
You won't be getting much of a backtrace out of `onefile` build, because it has a parent and a child process, and you are likely attaching the debugger to the parent.

**Reply by ilDeffo** [reply]
even without the onefile option, I can't get a lot of information. I tried to follow instructions at this link [](https://stackoverflow.com/questions/48278881/gdb-complaining-about-missing-raise-c/48287761) and the error message now is:
```
Program received signal SIGSEGV, Segmentation fault.
raise (sig=<optimized out>) at ../sysdeps/unix/sysv/linux/raise.c:50
50    return ret;
```

**Reply by rokm** [reply]
I think you're better off adding debug messages to your python code to see which part of the code triggers the segfault. And unless you can provide a minimum example that reproduces the problem, we can't really help you here.

**Reply by ilDeffo** [reply]
Looks like the problem was the library _darkdetect_ that i use to change colors in the UI according to the system theme...
